packages/parser/tests/lib/tsx.ts

```diff
@@ -1,115 +1,79 @@
-import { TSESLint } from '@typescript-eslint/experimental-utils';
-import * as parser from '../../src/parser';
+import { parseForESLint } from '../../src/parser';
+import { serializer } from '../tools/ts-error-serializer';
 
 //------------------------------------------------------------------------------
 // Tests
 //------------------------------------------------------------------------------
 
+expect.addSnapshotSerializer(serializer);
+
 describe('TSX', () => {
   describe("if the filename ends with '.tsx', enable jsx option automatically.", () => {
-    const linter = new TSESLint.Linter();
-    linter.defineParser('@typescript-eslint/parser', parser);
-
     it('filePath was not provided', () => {
       const code = 'const element = <T/>';
-      const config = {
-        parser: '@typescript-eslint/parser',
-      };
-      const messages = linter.verify(code, config);
-
-      expect(messages).toStrictEqual([
-        {
-          column: 18,
-          fatal: true,
-          line: 1,
-          message: "Parsing error: '>' expected.",
-          ruleId: null,
-          severity: 2,
-        },
-      ]);
+      try {
+        parseForESLint(code);
+      } catch (e) {
+        expect(e).toMatchSnapshot();
```

Comment:
could we use `toMatchInlineSnapshot` here instead of a separate file?
IMO inline snapshots are much better when the snapshots are that short and to-the-point.